tests/cnf/ran/oran/tests/oran-provision.go

```diff
@@ -1 +1,135 @@
+
+		By("verifying a NodePool was created")
+		nodePool, err := oran.PullNodePool(HubAPIClient, RANConfig.Spoke1Name, tsparams.HardwareManagerNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull NodePool for spoke 1")
+		Expect(nodePool.Object).ToNot(BeNil(), "Failed to get NodePool object for spoke 1")
+
+		By("verifying the ClusterInstance has the correct BMC details")
+		clusterInstance, err := siteconfig.PullClusterInstance(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull ClusterInstance for spoke 1")
+
+		clusterInstanceNode := clusterInstance.Definition.Spec.Nodes[0]
+		Expect(clusterInstanceNode.BmcAddress).
+			To(ContainSubstring(RANConfig.BMCHosts[0]), "ClusterInstance has incorrect BMC address")
+
+		bmcSecret, err := secret.Pull(HubAPIClient, clusterInstanceNode.BmcCredentialsName.Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull spoke 1 BMC secret")
+
+		bmcUsername, exists := bmcSecret.Definition.Data["username"]
+		Expect(exists).To(BeTrue(), "Username does not exist in BMC secret")
+
+		bmcUsername = bytes.TrimSpace(bmcUsername)
+		Expect(string(bmcUsername)).To(Equal(RANConfig.BMCUsername), "ClusterInstance has incorrect BMC username")
+
+		bmcPassword, exists := bmcSecret.Definition.Data["password"]
+		Expect(exists).To(BeTrue(), "Passowrd does not exist in BMC secret")
+
+		bmcPassword = bytes.TrimSpace(bmcPassword)
+		Expect(string(bmcPassword)).To(Equal(RANConfig.BMCPassword), "ClusterInstance has incorrect BMC password")
+
+		By("verifying spoke 1 pull-secret was created")
+		pullSecret, err := secret.Pull(HubAPIClient, "pull-secret", RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull pull-secret for spoke 1")
+		Expect(pullSecret.Object).ToNot(BeNil(), "Failed to get pull-secret object for spoke 1")
+
+		By("verifying spoke 1 extra-manifests was created")
+		extraManifests, err := configmap.Pull(HubAPIClient, tsparams.ExtraManifestsName, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull extra-manifests ConfigMap for spoke 1")
+		Expect(extraManifests.Object).ToNot(BeNil(), "Failed to get extra-manifests ConfigMap object for spoke 1")
+
+		By("verifying spoke 1 policy ConfigMap was created")
+		ztpNamespace := fmt.Sprintf("ztp-%s-%s", tsparams.ClusterTemplateName, RANConfig.ClusterTemplateAffix)
+		pgMap, err := configmap.Pull(
+			HubAPIClient, RANConfig.Spoke1Name+"-pg", ztpNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull policy ConfigMap for spoke 1")
+		Expect(pgMap.Object).ToNot(BeNil(), "Failed to get policy ConfigMap object for spoke 1")
+
+		By("verifying all the policies are compliant")
+		err = ocm.WaitForAllPoliciesComplianceState(
+			HubAPIClient, policiesv1.Compliant, time.Minute, runtimeclient.ListOptions{Namespace: RANConfig.Spoke1Name})
+		Expect(err).ToNot(HaveOccurred(), "Failed to verify all spoke 1 policies are compliant")
+
```

Comment:
As we are verifying multiple things post deployment, it would be better to not fail right away, but do all the checks then fail (error should be printed after each failure though), so if one fail, the rest of the verification won't be blocked.